code/datums/radiation_wave.dm

```diff
@@ -112,8 +116,14 @@
 			))
 		if(!can_contaminate || blacklisted[thing.type])
 			continue
-		if(prob(contamination_chance)) // Only stronk rads get to have little baby rads
-			if(SEND_SIGNAL(thing, COMSIG_ATOM_RAD_CONTAMINATING, strength) & COMPONENT_BLOCK_CONTAMINATION)
-				continue
-			var/rad_strength = (strength-RAD_MINIMUM_CONTAMINATION) * RAD_CONTAMINATION_STR_COEFFICIENT
+		if(CHECK_BITFIELD(thing.rad_flags, RAD_NO_CONTAMINATE) || SEND_SIGNAL(thing, COMSIG_ATOM_RAD_CONTAMINATING, strength) & COMPONENT_BLOCK_CONTAMINATION)
+			continue
+		contam_atoms += thing
+	var/did_contam = FALSE
+	if(length(can_contam))
+		var/rad_strength = ((strength-RAD_MINIMUM_CONTAMINATION) * RAD_CONTAMINATION_STR_COEFFICIENT)/contam_atoms.len
```

Comment:
The issue I see here is this is going to cause contamination amount to much more frequently be below the background level for radiation. This is much more a nerf than you think and the `RAD_CONTAMINATION_STR_COEFFICIENT` needs to be increased by a bit, probably to `0.3`

Reply:
Oh, it's not more a nerf than I think, there's a reason i called it "severe" in the title, and yeah, that's probably a good idea.